File: protocol/lua/ProtoFollow.lua
Proto4z.FOLLOW_INVALID = 0 
Proto4z.FOLLOW_DELETED = 1--路人 
Proto4z.FOLLOW_HATE = 2--黑 
Proto4z.FOLLOW_FOLLOWING = 3--粉 
 
Proto4z.FollowingInfo = {} --关注 
Proto4z.FollowingInfo.__getName = "FollowingInfo" 
Proto4z.FollowingInfo[1] = {name="following", type="ui64" }  
Proto4z.FollowingInfo[2] = {name="follower", type="ui64" }  
Proto4z.FollowingInfo[3] = {name="status", type="ui8" } --状态标志 
Proto4z.FollowingInfo[4] = {name="makeTime", type="ui32" } --建立时间 
Proto4z.FollowingInfo[5] = {name="lastChanged", type="ui32" } --最后更改时间 
 
Proto4z.FollowingInfoArray = {}  
Proto4z.FollowingInfoArray.__getName = "FollowingInfoArray" 
Proto4z.FollowingInfoArray.__getDesc = "array" 
Proto4z.FollowingInfoArray.__getTypeV = "FollowingInfo" 
 
Proto4z.register(1000,"GetFollowerReq") 
Proto4z.GetFollowerReq = {} --获取粉丝列表 
Proto4z.GetFollowerReq.__getID = 1000 
Proto4z.GetFollowerReq.__getName = "GetFollowerReq" 
Proto4z.GetFollowerReq[1] = {name="uID", type="ui64" }  
Proto4z.GetFollowerReq[2] = {name="status", type="ui8" } --所有/路人/黑/粉 
 
Proto4z.register(1001,"GetFollowerAck") 
Proto4z.GetFollowerAck = {} --获取粉丝列表 
Proto4z.GetFollowerAck.__getID = 1001 
Proto4z.GetFollowerAck.__getName = "GetFollowerAck" 
Proto4z.GetFollowerAck[1] = {name="status", type="ui8" } --所有/路人/黑/粉 
Proto4z.GetFollowerAck[2] = {name="follower", type="FollowingInfoArray" }  
 
Proto4z.register(1010,"GetFollowingReq") 
Proto4z.GetFollowingReq = {} --获取关注列表 
Proto4z.GetFollowingReq.__getID = 1010 
Proto4z.GetFollowingReq.__getName = "GetFollowingReq" 
Proto4z.GetFollowingReq[1] = {name="uID", type="ui64" }  
Proto4z.GetFollowingReq[2] = {name="status", type="ui8" } --所有/路人/黑/粉 
 
Proto4z.register(1011,"GetFollowingAck") 
Proto4z.GetFollowingAck = {} --获取关注列表 
Proto4z.GetFollowingAck.__getID = 1011 
Proto4z.GetFollowingAck.__getName = "GetFollowingAck" 
Proto4z.GetFollowingAck[1] = {name="status", type="ui8" } --所有/路人/黑/粉 
Proto4z.GetFollowingAck[2] = {name="following", type="FollowingInfoArray" }  
 
Proto4z.register(1008,"PokeReq") 
Proto4z.PokeReq = {} --操作 
Proto4z.PokeReq.__getID = 1008 
Proto4z.PokeReq.__getName = "PokeReq" 
Proto4z.PokeReq[1] = {name="dstStatus", type="ui8" } --路人/黑/粉 
Proto4z.PokeReq[2] = {name="following", type="ui64" } --关注目标ID 
 
Proto4z.register(1009,"PokeAck") 
Proto4z.PokeAck = {} --操作 
Proto4z.PokeAck.__getID = 1009 
Proto4z.PokeAck.__getName = "PokeAck" 
Proto4z.PokeAck[1] = {name="follow", type="FollowingInfo" }  

File: protocol/CSharp/ProtoFollow.cs
 
namespace Proto4z  
{ 
 
    public enum FOLLOW_STATUS : byte 
    { 
        FOLLOW_INVALID = 0,  
        FOLLOW_DELETED = 1, //路人  
        FOLLOW_HATE = 2, //黑  
        FOLLOW_FOLLOWING = 3, //粉  
    }; 
 
    public class FollowingInfo: Proto4z.IProtoObject //关注  
    {     
        //members   
        public ulong following;  
        public ulong follower;  
        public byte status; //状态标志  
        public uint makeTime; //建立时间  
        public uint lastChanged; //最后更改时间  
        public FollowingInfo()  
        { 
            following = 0;  
            follower = 0;  
            status = 0;  
            makeTime = 0;  
            lastChanged = 0;  
        } 
        public FollowingInfo(ulong following, ulong follower, byte status, uint makeTime, uint lastChanged) 
        { 
            this.following = following; 
            this.follower = follower; 
            this.status = status; 
            this.makeTime = makeTime; 
            this.lastChanged = lastChanged; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.following)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.follower)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.status)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI32(this.makeTime)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI32(this.lastChanged)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.following = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.follower = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.status = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            this.makeTime = Proto4z.BaseProtoObject.decodeUI32(binData, ref pos); 
            this.lastChanged = Proto4z.BaseProtoObject.decodeUI32(binData, ref pos); 
            return pos; 
        } 
    } 
 
 
    public class FollowingInfoArray : System.Collections.Generic.List<FollowingInfo>, Proto4z.IProtoObject  
    { 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var ret = new System.Collections.Generic.List<byte>(); 
            int len = (int)this.Count; 
            ret.AddRange(Proto4z.BaseProtoObject.encodeI32(len)); 
            for (int i = 0; i < this.Count; i++ ) 
            { 
                ret.AddRange(this[i].__encode()); 
            } 
            return ret; 
        } 
 
        public int __decode(byte[] binData, ref int pos) 
        { 
            int len = Proto4z.BaseProtoObject.decodeI32(binData, ref pos); 
            if(len > 0) 
            { 
                for (int i=0; i<len; i++) 
                { 
                    var data = new FollowingInfo(); 
                    data.__decode(binData, ref pos); 
                    this.Add(data); 
                } 
            } 
            return pos; 
        } 
    } 
 
    public class GetFollowerReq: Proto4z.IProtoObject //获取粉丝列表  
    {     
        //proto id   
        public const ushort protoID = 1000;  
        static public ushort getProtoID() { return 1000; } 
        static public string getProtoName() { return "GetFollowerReq"; } 
        //members   
        public ulong uID;  
        public byte status; //所有/路人/黑/粉  
        public GetFollowerReq()  
        { 
            uID = 0;  
            status = 0;  
        } 
        public GetFollowerReq(ulong uID, byte status) 
        { 
            this.uID = uID; 
            this.status = status; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.uID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.status)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.uID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.status = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class GetFollowerAck: Proto4z.IProtoObject //获取粉丝列表  
    {     
        //proto id   
        public const ushort protoID = 1001;  
        static public ushort getProtoID() { return 1001; } 
        static public string getProtoName() { return "GetFollowerAck"; } 
        //members   
        public byte status; //所有/路人/黑/粉  
        public FollowingInfoArray follower;  
        public GetFollowerAck()  
        { 
            status = 0;  
            follower = new FollowingInfoArray();  
        } 
        public GetFollowerAck(byte status, FollowingInfoArray follower) 
        { 
            this.status = status; 
            this.follower = follower; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.status)); 
            if (this.follower == null) this.follower = new FollowingInfoArray(); 
            data.AddRange(this.follower.__encode()); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.status = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            this.follower = new FollowingInfoArray(); 
            this.follower.__decode(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class GetFollowingReq: Proto4z.IProtoObject //获取关注列表  
    {     
        //proto id   
        public const ushort protoID = 1002;  
        static public ushort getProtoID() { return 1002; } 
        static public string getProtoName() { return "GetFollowingReq"; } 
        //members   
        public ulong uID;  
        public byte status; //所有/路人/黑/粉  
        public GetFollowingReq()  
        { 
            uID = 0;  
            status = 0;  
        } 
        public GetFollowingReq(ulong uID, byte status) 
        { 
            this.uID = uID; 
            this.status = status; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.uID)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.status)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.uID = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            this.status = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class GetFollowingAck: Proto4z.IProtoObject //获取关注列表  
    {     
        //proto id   
        public const ushort protoID = 1003;  
        static public ushort getProtoID() { return 1003; } 
        static public string getProtoName() { return "GetFollowingAck"; } 
        //members   
        public byte status; //所有/路人/黑/粉  
        public FollowingInfoArray following;  
        public GetFollowingAck()  
        { 
            status = 0;  
            following = new FollowingInfoArray();  
        } 
        public GetFollowingAck(byte status, FollowingInfoArray following) 
        { 
            this.status = status; 
            this.following = following; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.status)); 
            if (this.following == null) this.following = new FollowingInfoArray(); 
            data.AddRange(this.following.__encode()); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.status = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            this.following = new FollowingInfoArray(); 
            this.following.__decode(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class PokeReq: Proto4z.IProtoObject //操作  
    {     
        //proto id   
        public const ushort protoID = 1004;  
        static public ushort getProtoID() { return 1004; } 
        static public string getProtoName() { return "PokeReq"; } 
        //members   
        public byte dstStatus; //路人/黑/粉  
        public ulong following; //关注目标ID  
        public PokeReq()  
        { 
            dstStatus = 0;  
            following = 0;  
        } 
        public PokeReq(byte dstStatus, ulong following) 
        { 
            this.dstStatus = dstStatus; 
            this.following = following; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI8(this.dstStatus)); 
            data.AddRange(Proto4z.BaseProtoObject.encodeUI64(this.following)); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.dstStatus = Proto4z.BaseProtoObject.decodeUI8(binData, ref pos); 
            this.following = Proto4z.BaseProtoObject.decodeUI64(binData, ref pos); 
            return pos; 
        } 
    } 
 
    public class PokeAck: Proto4z.IProtoObject //操作  
    {     
        //proto id   
        public const ushort protoID = 1005;  
        static public ushort getProtoID() { return 1005; } 
        static public string getProtoName() { return "PokeAck"; } 
        //members   
        public FollowingInfo follow;  
        public PokeAck()  
        { 
            follow = new FollowingInfo();  
        } 
        public PokeAck(FollowingInfo follow) 
        { 
            this.follow = follow; 
        } 
        public System.Collections.Generic.List<byte> __encode() 
        { 
            var data = new System.Collections.Generic.List<byte>(); 
            if (this.follow == null) this.follow = new FollowingInfo(); 
            data.AddRange(this.follow.__encode()); 
            return data; 
        } 
        public int __decode(byte[] binData, ref int pos) 
        { 
            this.follow = new FollowingInfo(); 
            this.follow.__decode(binData, ref pos); 
            return pos; 
        } 
    } 
 
} 
 
 
